src/OrderInterface.php

```diff
@@ -0,0 +1,15 @@
+<?php
+
+namespace Dhii\Storage\Resource\Sql;
+
+/**
+ * Something that describes how SQL results should be ordered.
+ *
+ * @since [*next-version*]
+ */
+interface OrderInterface extends
```

Comment:
How about extending `EntityFieldAwareInterface`, forming a "has" relationship with the field?

Reply:
Perhaps, a better approach is to keep it aware of the field and entity separately, like it is now. This means that there can be an implementation that uses an entity field internally, but this is not required, allowing just simple strings.